autoPyTorch/optimizer/smbo.py

```diff
@@ -0,0 +1,325 @@
+from autoPyTorch.pipeline.components.training.metrics.base import autoPyTorchMetric
+from autoPyTorch.utils.backend import Backend
+from autoPyTorch.utils.logging_ import get_named_client_logger
+from autoPyTorch.utils.stopwatch import StopWatch
+
+
+def get_smac_object(
+    scenario_dict: typing.Dict[str, typing.Any],
+    seed: int,
+    ta: typing.Callable,
+    ta_kwargs: typing.Dict[str, typing.Any],
+    n_jobs: int,
+    dask_client: typing.Optional[dask.distributed.Client],
+) -> SMAC4AC:
+    """
+    This function returns an SMAC object that is gonna be used as
+    optimizer of pipelines
+
+    Args:
+        scenario_dict (typing.Dict[str, typing.Any]): constrain on how to run
+            the jobs
+        seed (int): to make the job deterministic
+        ta (typing.Callable): the function to be intensifier by smac
+        ta_kwargs (typing.Dict[str, typing.Any]): Arguments to the above ta
+        n_jobs (int): Amount of cores to use for this task
+        dask_client (dask.distributed.Client): User provided scheduler
+
+    Returns:
+        (SMAC4AC): sequential model algorithm configuration object
+
+    """
+    if len(scenario_dict['instances']) > 1:
+        intensifier = Intensifier
+    else:
+        intensifier = SimpleIntensifier
+
+    rh2EPM = RunHistory2EPM4LogCost
+    return SMAC4AC(
+        scenario=Scenario(scenario_dict),
+        rng=seed,
+        runhistory2epm=rh2EPM,
+        tae_runner=ta,
+        tae_runner_kwargs=ta_kwargs,
+        initial_configurations=None,
+        run_id=seed,
+        intensifier=intensifier,
+        dask_client=dask_client,
+        n_jobs=n_jobs,
+    )
+
+
+class AutoMLSMBO(object):
+
+    def __init__(self,
+                 config_space: ConfigSpace.ConfigurationSpace,
+                 dataset_name: str,
+                 backend: Backend,
+                 total_walltime_limit: float,
+                 func_eval_time_limit: float,
+                 memory_limit: typing.Optional[int],
+                 metric: typing.List[autoPyTorchMetric],
+                 watcher: StopWatch,
+                 n_jobs: int,
+                 dask_client: typing.Optional[dask.distributed.Client],
+                 start_num_run: int = 1,
+                 seed: int = 1,
+                 resampling_strategy: str = 'holdout',
+                 resampling_strategy_args: typing.Optional[typing.Dict[str, typing.Any]] = None,
+                 include: typing.Optional[typing.Dict[str, typing.Any]] = None,
+                 exclude: typing.Optional[typing.Dict[str, typing.Any]] = None,
+                 disable_file_output: bool = False,
+                 smac_scenario_args: typing.Optional[typing.Dict[str, typing.Any]] = None,
+                 get_smac_object_callback: typing.Optional[typing.Callable] = None,
+                 # TODO: Re-enable when ensemble merged
+                 # ensemble_callback: typing.Optional[EnsembleBuilderManager] = None,
```

Comment:
We can turn this on to see if everything is working. When you are done with the changes I can do a PR on your PR to enable this

Reply:
Sure